### PR TITLE
[glyphdata] handle production names for ligatures with script suffixes

### PR DESCRIPTION
**Pull request by khaledhosny**
We had them handled in category detection but not in production names.
This change generalizes this and adds few more conditions:

1. Don’t add script suffix to ligature part that already have one.
2. Don’t add it if the part had an entry in GlyphData (to handle cases like "brevecomb_Dboldscript-math").

Should fix https://github.com/googlefonts/glyphsLib/issues/770

### CODE REVIEW

**Comment by khaledhosny** [comment]
@madig feel free to add more tests. This probably does not fix all the differences you are seeing.

**Comment by khaledhosny** [comment]
The regression tests failure is because we now detect proper names for more ligatures.

**Comment by madig** [comment]
That was fast. Thanks! A ripgrep and sort through the diffs shows all new names are uni-names now, so that's good.

I added more testcases and found several diffs though. Not entirely sure if some of it is due to outdated info in GlyphData.xml? I used Glyphs 2.6.6 to generated the expected values.

**Comment by khaledhosny** [comment]
With ef6fed2 the failures are down to 16 (was 70).

**Comment by khaledhosny** [comment]
The last commit should have fixed a punch of failures, but it does not. We now generate uni names for the affected glyphs, but they are different from Glyphs’.

**Comment by khaledhosny** [comment]
Remaining failures:
```
FAILED tests/glyphdata_test.py::test_prod_names[sh_r-deva-uni0936094D0930094D] - AssertionError: assert 'uni0936094D094D0930' == 'uni0936094D0930094D'
FAILED tests/glyphdata_test.py::test_prod_names[idotaccent.sc-i.sc.loclTRK] - AssertionError: assert 'i.loclTRK.sc' == 'i.sc.loclTRK'
FAILED tests/glyphdata_test.py::test_prod_names[t_r-deva-uni0924094D0930094D] - AssertionError: assert 'uni0924094D094D0930' == 'uni0924094D0930094D'
```

I think we should just accept the `idotaccent.sc` one as insignificant difference, but I have no idea what is causing the difference in the other two; we are splitting the ligatures correctly and getting the correct part names from GlyphData and I have no idea from where Glyphs is pulling the difference (AFAICT the relavant entries are the same in our copy of the data and Glyphs’).

**Comment by anthrotype** [comment]
what's the status of this PR?

**Comment by schriftgestalt** [comment]
I’m reworking the glyph data handling quite a bit right now so that is probably not needed any more.

**Comment by anthrotype** [comment]
> I’m reworking the glyph data handling quite a bit right now so that is probably not needed any more.

how's that going?

**Comment by khaledhosny** [comment]
I updated the expectations of the three remaining test failures. These should be investigated, but the failures are unrelated to splitting ligatures, we are doing this correctly but Glyphs is doing something more here.

If there are no objections, I’m going to merge this.

**Comment by khaledhosny** [comment]
(As said previously, the regression failures are this PR working and handling more ligatures than before).

**Comment by anthrotype** [comment]
maybe instead of making the test pass you could mark them as `xfail`?

File: tests/glyphdata_test.py [hunk 1]
```diff
@@ -190,5 +193,221 @@ def test_glyphdata_no_duplicates(self):
...
+    "ro-khmer.pre.narrow": "uni17D2179A.narrow",
+    "rra_iMatra-tamil": "uni0BB10BBF",
+    "rra_iiMatra-tamil": "uni0BB10BC0",
+    "rra_uMatra-tamil": "uni0BB10BC1",
+    "rra_uuMatra-tamil": "uni0BB10BC2",
+    "sa_iMatra-tamil": "uni0BB80BBF",
+    "sa_iiMatra-tamil": "uni0BB80BC0",
+    "sa_uMatra-tamil": "uni0BB80BC1",
+    "sa_uuMatra-tamil": "uni0BB80BC2",
+    "sh_r-deva": "uni0936094D094D0930",  # uni0936094D0930094D
```
Comment (anthrotype) [inline tests/glyphdata_test.py:372]: i'm not familiar with devanagari conjucts but looks like there's some reordering that is (or is not) happening here?
/cc @schriftgestalt 

```diff
- uni0936094D094D0930
+ uni0936094D0930094D
```

**Comment by anthrotype** [comment]
+1 to merge this as improves current handling of ligatures' production names; as for any outstanding issues left, we could track them in their own issue and tackle later

**Comment by schriftgestalt** [comment]
That should be handle with my pull request that was never excepted.

**Comment by khaledhosny** [comment]
> That should be handle with my pull request that was never excepted.

Which one, https://github.com/googlefonts/glyphsLib/pull/818?

If you want to finish it in some reasonable time frame, I’m more than happy to close this one instead, otherwise I’m currently blocked on this issue and need a fix ASAP.

**Comment by schriftgestalt** [comment]
It was mergable when opened the pull request.

**Comment by schriftgestalt** [comment]
My PR fixes and improves several areas (names, indo, writing direction …). Adding more changes now will make it even more complicated to merge it. 
And that code is needed also for round tripping Glyphs 2 and 3 files.

**Comment by khaledhosny** [comment]
So what is the resolution here? #818 is unmergable as it stands and I’m still blocked.

**Comment by schriftgestalt** [comment]
I'll panned to work on glyphsLib next week. Finishing the Glyphs3 branch and rebasing the info branch.

**Comment by khaledhosny** [comment]
I hate to keep saying this, but this PR have been open for over a tear and half and it is been 3 weeks since I rebased it, and I’m still blocked. I suggest merging this and I’ll try to help with rebasing #818 and solving the conflicts afterwords.

File: tests/glyphdata_test.py (hunk 1 above)
Comment (khaledhosny) [reply tests/glyphdata_test.py:372]: I think this is addressed by the mini Devanagri shaper in #818, so I’m not going to worry about it here.

**Comment by anthrotype** [comment]
Feel free to merge this as soon as you make the CI pass, thanks

**Comment by khaledhosny** [comment]
BTW, I tried to rebase #818 on current master, and there are significant conflicts, so I don’t think merging this is going to make it much worse.

**Pull request merged**